api/src/paths/projects/{projectId}/artifacts/{fileName}/download.ts

```diff
@@ -0,0 +1,108 @@
+'use strict';
```

Comment:
Instead of creating a folder called `{fileName}` and having a file inside called `download.ts` you can actually just have a file called `{fileName}.ts`

So the structure would be: `paths/projects/{projectId}/artifacts/{fileName}.ts`

Reply:
The content of the file would be the same (other than adjusting the relative import paths)

Reply:
Take a look at HATEOAS.  I was thinking it would be nice to adopt this style, where the action is placed at the end of the path.
 https://en.wikipedia.org/wiki/HATEOAS#:~:text=Hypermedia%20as%20the%20Engine%20of,provide%20information%20dynamically%20through%20hypermedia.
 
Happy to change it back.

Reply:
Hmm, interesting.  I've never heard of that concept by name, though I've made apis with that sort of thing.  Usually it would come up because of some non-standard actions like "publish/unpublish".  

Okay, well lets leave it as you have it for now then.